misc: convert TR1 cutscenes to TR2 format

lahm86 opened the pull request:
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TRX/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change
- [x] I have added a readme entry about my new feature or OG bug fix, or it is a different change

#### Description

This updates TR1 cutscene data to make sure that Lara (as `O_LARA`) is always present, meaning the logic can be aligned to TR2 in terms of setup, control and camera behaviour. This brings in a few changes as a result:
- Lara's shadow is now visible in cutscenes
- Lara's lighting in the Atlantis cutscene is slightly different
- Mesh swap, flip map, and camera pos/angle game flow sequence events are now redundant - everything feeds off the data (TR2 as well, no data changes needed)
- Lara's braid injection is much simplified as the cutscene meshes don't need specific handling

Lara's setup and control remain separate for now to allow for testing, although these are practically identical so that should be an easy next task.

Ref: https://github.com/LostArtefacts/TRXInjectionTool/pull/141

lahm86 committed:
misc: convert TR1 cutscenes to TR2 format

This updates TR1 cutscenes via injections such that O_LARA is always
present, and hence allows aligning the logic to match TR2. Braid
injections are also simplified as the injected data standardizes
Lara's meshes. In the Natla-only cutscene, Lara is still present,
only hidden. Flip map and mesh swap game flow commands are replaced
with animation flip effects throughout.

lahm86 committed:
t2/cutscene: set camera angle from Lara's position

lahm86 committed:
game-flow: remove redundant sequence types

This removes flip map, mesh swap, and camera position/angle sequence
types as these are all now well-defined in the data for both
games. Also removes lara_type from GF levels as this is no longer
required for the braid in TR1.

rr- reviewed (commented):
Thank you – this looks fantastic 🩷

However, I'd either enable the shadows for all cutscene players, or disable it in TR1, as the Larson cutscene looks unfortunate. It should be possible to target it with `GF_GetCurrentLevel()->type == GFL_CUTSCENE`. LMK WDYT?

lahm86 commented:
> Thank you – this looks fantastic 🩷
> 
> However, I'd either enable the shadows for all cutscene players, or disable it in TR1, as the Larson cutscene looks unfortunate. It should be possible to target it with `GF_GetCurrentLevel()->type == GFL_CUTSCENE`. LMK WDYT?

How about a new flip effect (item action) to switch it off? That way TR2's shadow for Lara in cutscene remains, but we can get rid in TR1. The problem with the other actors is we'd need to define shadow size and they're of course different across the files.

rr- commented:
On that end, maybe we could make the flip effect to set the shadow size to make it more versatile? And setting it to 0 disables it completely.

lahm86 commented:
That would be ideal, but flip effects don't have scope for args unfortunately (it'd need an entirely new animation command). Maybe the on/off support would be OK for now and in the future Lua could manage `obj->shadow_size`.

Added that change for now, LMK if OK.

lahm86 committed:
items/actions: add shadow control flip effects

This adds new flip effects to allow enabling or disabling an item's
shadow. This is used in TR1 cutscenes to disable Lara's shadow, per OG.

rr- reviewed (approved):
LGTM and thank you 😌
